### PR TITLE
🔨 forcibly removing DOI citation lookup & insertion

### PR DESCRIPTION
**Pull request by stevejpurves**
Trying to address issue #196 here.

Currently, this is intended for discussion on how to best provide an option in frontmatter.

### CODE REVIEW

File: packages/myst-cli/src/process/mdast.ts [hunk 1]
```diff
@@ -136,13 +136,13 @@ export async function transformMdast(
     new WikiTransformer(),
     new GithubTransformer(),
     new RRIDTransformer(),
-    new DOITransformer(), // This also is picked up in the next transform
+    // new DOITransformer(), // This also is picked up in the next transform
```
Comment (stevejpurves) [inline packages/myst-cli/src/process/mdast.ts:139]: Removing this here actually seems ot have no effect. What is the role of this transformer?

File: packages/myst-cli/src/process/mdast.ts [hunk 2]
```diff
@@ -136,13 +136,13 @@ export async function transformMdast(
...
     new MystTransformer(intersphinx),
   ];
   linksTransform(mdast, vfile, { transformers, selector: LINKS_SELECTOR });
 
   // Initialize citation renderers for this (non-bib) file
-  cache.$citationRenderers[file] = await transformLinkedDOIs(log, mdast, cache.$doiRenderers, file);
+  // cache.$citationRenderers[file] = await transformLinkedDOIs(log, mdast, cache.$doiRenderers, file);
```
Comment (stevejpurves) [inline packages/myst-cli/src/process/mdast.ts:145]: removing the renderer here stopped lookup and rendering of `\cite{}` commands.

**Comment by stevejpurves** [comment]
@fwkoch so would adding a frontmatter field, and checking that in these two locations, adding the transformers only when `true` be the best way forward?

```
convert_doi_links: true|false
```
?

**Comment by rowanc1** [comment]
We can certainly come back to this! My main hesitation at this stage is we don't have any centralized options yet for controlling parse behaviour. I think this is an obvious thing we need to add, but it will take some work. :)

**Pull request closed without merging**